### PR TITLE
Modify Queue already add to User not change bug --  fix #1514

### LINKED ISSUE #1514: the field queue in  table t_ds_user not change with tabke t_ds_queue modify queue_name field
**Describe the bug**
A user have add a queue like 'developQueue', when admin modify the  developQueue queue, the 
field queue in  table t_ds_queue not change

**To Reproduce**
1. Add a user with a queue like  'developQueue'
2. admin modify the 'developQueue' queue
3. the field queue in  table t_ds_user not change

**Expected behavior**
 the field queue in  table t_ds_user change with tabke t_ds_queue any modify

**Screenshots**

**Which version of Dolphin Scheduler:**
 -[dev]

**Requirement or improvement
- the field queue in  table t_ds_user change with table t_ds_queue modify the field queue_name

### PR DESCRIPTION
**Pull request by Yeleights**
#1514  What is the purpose of the pull request

*fix bug: the field queue in table t_ds_user does not change with table t_ds_queue modify the field queue_name*

## Brief change log
  - *modify api/service/QueueService.java*
  - *modify dao/mapper/UserMapper.java*
  - *modify dao/mapper/UserMapper.xml*

## Verify this pull request
This pull request is already covered by existing tests QueueControllerTest

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/QueueService.java [hunk 1]
```diff
@@ -240,21 +252,35 @@ public Result verifyQueue(String queue, String queueName) {
...
+
+    /**
+     * check old queue name using by any user
+     * if need to update user
+     *
+     * @param queueOld old queue name
+     * @param queueNew new queue name
+     * @return true if need to update user
+     */
+    private boolean checkUserRelatedQueue(String queueOld, String queueNew) {
```
Comment (Technoboy-) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/QueueService.java]: oldQueue， newQueue

Comment (Technoboy-) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/QueueService.java]: method name is not readable.     checkIfQueueIsInUsing is prefered

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/UserMapper.java [hunk 2]
```diff
@@ -95,4 +95,18 @@
...
+     */
+    List<User> queryUserListByQueue(@Param("queueName") String queueName);
+
+    /**
+     * update user with old queue
+     * @param queueOld old queue name
+     * @param queueNew new queue name
+     * @return update rows
+     */
+    Integer updateUserQueue(@Param("queueOld") String queueOld, @Param("queueNew") String queueNew);
```
Comment (Technoboy-) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/UserMapper.java]: oldQueue, newQueue

Comment (Yeleights) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/mapper/UserMapper.java]: Method and param name have been modified, please review. thanks~

**Pull request merged**